app/src/main/java/fr/free/nrw/commons/upload/MultipleShareActivity.java

```diff
@@ -233,6 +237,43 @@ protected void onCreate(Bundle savedInstanceState) {
         }
     }
 
+    private void initPermissionsRationaleDialog() {
+        if (storagePermissionInfoDialog == null) {
+            storagePermissionInfoDialog = DialogUtil
+                    .getAlertDialogWithPositiveAndNegativeCallbacks(
+                            MultipleShareActivity.this,
+                            getString(R.string.storage_permission), getString(
+                                    R.string.write_storage_permission_rationale_for_image_share),
+                            R.drawable.ic_launcher, new Callback() {
+                                @Override
+                                public void onPositiveButtonClicked() {
+                                    //If the user is willing to give us the permission
+                                    //But had somehow previously choose never ask again, we take him to app settings to manually enable permission
+                                    if(null== permissionDeniedResponse){
+                                        //Dexter returned null, lets see if this ever happens
+                                        return;
+                                    }
+                                    else if (permissionDeniedResponse.isPermanentlyDenied()) {
+                                        PermissionUtils
+                                                .askUserToManuallyEnablePermissionFromSettings(
```

Comment:
Generally it is not necessary to split these lines, as per #465

Reply:
Agreed..will do